test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -862,13 +928,15 @@ void UcxConnection::request_completed(ucx_request *r)
 
 void UcxConnection::handle_connection_error(ucs_status_t status)
 {
+    if (UCS_STATUS_IS_ERR(_ucx_status)) {
+        return;
+    }
+
     UCX_CONN_LOG << "detected error: " << ucs_status_string(status);
-    _ucx_status = status;
+    _ucx_status   = status;
 
-    if (_connected) {
```

Comment:
why removed this check?

Reply:
the failed connection goes thru common failed list

Reply:
do we avoid calling user callback if _connected is false?
if not, why not, since it was the case before this PR?

Reply:
before this PR we handled failed connection before and after "connected" differently.
now `_connected` is needed only to handle active servers list on the app level

Reply:
```c
void UcxContext::progress_failed_connections()
{
    while (!_failed_conns.empty()) {
        UcxConnection *conn = _failed_conns.front();
        _failed_conns.pop_front();
        if (conn->is_established) {
            dispatch_connection_error(conn);
        } else {
            conn->callback(error);
        }
    }
}
```